### PR TITLE
Add metrics for the forward handler

### PR DESCRIPTION
**Pull request by xichen2020**
cc @cw9 @prateek @jeromefroe @schallert @dgromov 

This PR adds metrics for the forward handler responsible for forwarding metrics from the agg tier to the downstream services.

### CODE REVIEW

File: aggregator/handler/forward.go [hunk 1]
```diff
@@ -53,6 +56,34 @@ type tryConnectFn func(addr string) (*net.TCPConn, error)
...
+	queueSize        tally.Gauge
+}
+
+func newForwardHandlerMetrics(scope tally.Scope) forwardHandlerMetrics {
+	return forwardHandlerMetrics{
+		connectTimeouts:  scope.Counter("connect-timeouts"),
+		connectSuccesses: scope.Counter("connect-successes"),
+		enqueueErrors:    scope.Counter("enqueue-errors"),
+		enqueueSuccesses: scope.Counter("enqueue-successes"),
+		currentDropped:   scope.Counter("current-dropped"),
```
Comment (cw9) [inline aggregator/handler/forward.go]: nit: latestDropped?

File: aggregator/handler/forward.go [hunk 2]
```diff
@@ -61,17 +92,20 @@ type forwardHandler struct {
...
-	bufCh             chan msgpack.Buffer
-	tryConnectFn      tryConnectFn
-	dialWithTimeoutFn dialWithTimeoutFn
-	writeToConnFn     writeToConnFn
-	wg                sync.WaitGroup
-	closed            bool
+	bufCh              chan msgpack.Buffer
+	wg                 sync.WaitGroup
+	closed             bool
+	doneCh             chan struct{}
```
Comment (cw9) [inline aggregator/handler/forward.go]: nit: closedCh?

File: aggregator/handler/forward.go (hunk 1 above)
Comment (xichen2020) [inline aggregator/handler/forward.go]: Sure thing

File: aggregator/handler/forward.go (hunk 2 above)
Comment (xichen2020) [inline aggregator/handler/forward.go]: Sounds good

File: aggregator/handler/forward.go (hunk 1 above)
Comment (xichen2020) [inline aggregator/handler/forward.go]: Actually it's not the latest, rather the current buffer is the oldest, i think i'll stick to this.

**Pull request merged**